File: Java/Medium/Accepted/maximum-number-of-events-that-can-be-attended_1.java
class Solution {
    public int maxEvents(int[][] events) {
        
        Arrays.sort(events, (a, b) -> a[0] - b[0]);
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int i = 0;
        int res = 0;
        for (int d = 1; d <= 100000; d++) {
            while (i < events.length && events[i][0] == d) {
                pq.add(events[i++][1]);
            }
            while (!pq.isEmpty() && pq.peek() < d) {
                pq.poll();
            }
            if (!pq.isEmpty()) {
                pq.poll();
                res++;
            }
        }
        return res;
    }
}

File: C++/Medium/Accepted/maximum-number-of-events-that-can-be-attended_1.cpp
class Solution {
public:
    int maxEvents(vector<vector<int>>& events) {
        
        sort(events.begin(), events.end());
        int n = events.size();
        priority_queue<int, vector<int>, greater<int>> pq;
        int res = 0;
        int i = 0;
        for (int d = 1; d <= 100000; d++) {
            while (!pq.empty() && pq.top() < d) pq.pop();
            while (i < n && events[i][0] == d) pq.push(events[i++][1]);
            if (!pq.empty()) {
                pq.pop();
                res++;
            }
        }
        return res;
    }
};